Proteomics/Protein/Protein.cs

```diff
@@ -585,60 +585,236 @@ public void RestoreUnfilteredModifications()
             return validModDictionary;
         }
 
-        public void AddBiomarkersToProteolysisProducts(int fullProteinOneBasedBegin, int fullProteinOneBasedEnd, bool addNterminalDigestionBiomarkers, bool addCterminalDigestionBiomarkers, bool retainNterminalMethionine, int minProductBaseSequenceLength, int lengthOfProteolysis, string proteolyisisProductName)
+        public void AddBiomarkersToProteolysisProducts(int fullProteinOneBasedBegin, int fullProteinOneBasedEnd, bool addNterminalDigestionBiomarkers, bool addCterminalDigestionBiomarkers, InitiatorMethionineBehavior initiatorMethionineBehavior, int minProductBaseSequenceLength, int lengthOfProteolysis, string proteolyisisProductName)
```

Comment:
It took me a minute to figure out how initiator methionine are being dealt with by the logic. Could you add a comment specifically pointing out that the fullProteinOneBasedBegin + 1 is the part that's skipping the methionine?

Reply:
ha. it is not skipping the methionine, the n-terminus all by itself is index position 0, regardless of if ther is methionine at pos 1 or not

Reply:
Yeah that makes complete sense to me now.